jenkinsfiles/cordova_jenkinsfile.groovy

```diff
@@ -1,14 +1,72 @@
 /**
 * Cordova Jenkinsfile
 */
-node('android') {
-   stage 'Checkout'
-	 checkout scm
 
-	 stage 'Prepare'
-	 cordova platform add android
-	 cordova prepare
+def platform = params?.PLATFORM?.trim()
+def buildType = params?.BUILD_CONFIG?.trim()
+def keystore_path = "/opt/android-sdk-linux/android.debug"
 
-	 stage 'Build'
-	 cordova build android
+if(platform == 'android') {
+    node(platform) {
+        stage("Checkout") {
+            checkout scm
+        }
+
+        stage("Prepare") {
+            sh "cordova platform add $platform"
+            sh "cordova prepare $platform"
+            if (buildType == 'debug') {
+	            file_content= sh (
+	                script: 'jq --arg test_path /opt/android-sdk-linux/android.debug \'.android.debug.keystore = $test_path\' -r build.json',
+	                returnStdout: true
+	                )
+            	sh "echo '${file_content}' > build.json"
+            }else{
+	            file_content= sh (
+	                script: 'jq --arg test_path /opt/android-sdk-linux/android.debug \'.android.debug.keystore = $test_path\' -r build.json',
+	                returnStdout: true
+	                )
+		        sh "echo '${file_content}' > build.json"
+            }
+
+            sh "cat build.json"
+        }
+
+        stage("Build") {
+            sh 'cordova build $platform --buildConfig build.json'
+        }
+
+        stage("Archive") {
+            if (buildType == 'release') {
+                archiveArtifacts artifacts: 'platforms/android/build/outputs/apk/android-release.apk', excludes: 'platforms/android/build/outputs/apk/*-unaligned.apk'
+            } else {
+                archiveArtifacts artifacts: 'platforms/android/build/outputs/apk/android-debug.apk', excludes: 'platforms/android/build/outputs/apk/*-unaligned.apk'
+            }
+        }
+    }
+}
+
+if (platform == 'ios') {
```

Comment:
yes, I was thinking in moving the platform verification part to the stages that need this verification (prepare and archive)